app.vue

```diff
@@ -216,12 +216,29 @@ useHead({
       return bodyClass.join(' ')
     }),
   },
-  script: [
-    {
-      innerHTML: `if('serviceWorker' in navigator){window.addEventListener('load', () => {navigator.serviceWorker.register('/sw.js', { scope: '/' })})}`,
-    },
-  ],
 })
+const hasUpgrade = ref<boolean>(false)
```

Comment:
Let's name it `hasServiceWorkerUpgrade` and move to the top where all refs sit.

Also qq, would that also work if we wrap the below code into function and call in `onMounted` hook as `setupServiceWorker`?